### PR TITLE
Publish nightly version to Docker Hub too

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

With https://github.com/ChainSafe/lodestar/pull/2647 we are releasing nightly versions to NPM. We should do the same for Docker Hub.

**Description**

Grab the tag from lerna's logic and publish to Docker Hub.

### CODE REVIEW

File: .github/workflows/release-nightly.yml [hunk 1]
```diff
@@ -40,3 +40,24 @@ jobs:
         run: yarn run publish:nightly
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Get latest tag
+        uses: actions-ecosystem/action-get-latest-tag@v1
```
Comment (mpetrunic) [inline .github/workflows/release-nightly.yml]: ain't this loading latest git tag?

Nightly releases don't have tag.

You could do something like:
```
runs: |
   packageJson=`cat ./package.json`
   version=`echo $(jq -r '.version' <<< "$packageJson")`
   echo "::set-output name=version::$version"
```
not sure if this works needs testing 😅

File: .github/workflows/release-nightly.yml [hunk 2]
```diff
@@ -40,3 +40,24 @@ jobs:
...
+    runs-on: ubuntu-latest
+    needs: publish
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      # TODO: Prevent Docker Hub tagging this image as "latest", tag as "next" instead
```
Comment (mpetrunic) [inline .github/workflows/release-nightly.yml]: just add 
`docker tag chainsafe/lodestar:${{ needs.publish.outputs.tag }} chainsafe/lodestar:next`
and also push chainsafe/lodestar:next

**Review by mpetrunic — COMMENTED** [review]
I think it would be wise to also change dockerfile to only install npm package otherwise we will spend huge amount of ci time just building that docker images

File: .github/workflows/release-nightly.yml (hunk 2 above)
Comment (dapplion) [reply .github/workflows/release-nightly.yml]: Thanks! What about preventing dockerhub from tagging it as latest?

**Comment by dapplion** [comment]
> I think it would be wise to also change dockerfile to only install npm package otherwise we will spend huge amount of ci time just building that docker images

Yeah I'm on it https://github.com/ChainSafe/lodestar/pull/2672

File: .github/workflows/release-nightly.yml (hunk 2 above)
Comment (mpetrunic) [reply .github/workflows/release-nightly.yml]: I don't think dockerhub does that by default. You have to push latest tag in order to update it or push image without tag

Comment (mpetrunic) [reply .github/workflows/release-nightly.yml]: or not hmm
We should probably start pushing stable tag with our main release

**Comment by dapplion** [comment]
Warning! There can be a race condition if NPM does not serve the recently pushed version immediately. The build should target the exact version of this nightly build.

We could use this API https://registry.npmjs.org/@chainsafe/lodestar-cli to ensure that our target nightly version is already available.

**Review by mpetrunic — DISMISSED** [review]

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**